### PR TITLE
add differences between mintable xc20s and erc20s

### PR DESCRIPTION
**Pull request by eshaben**
### Description

add differences between mintable xc20s and erc20s

### Checklist

- [ ] If this requires translations for the `moonbeam-docs-cn` repo, I have created a ticket for the translations in Jira

### After Translation Requirements

- [ ] Will need to create PR in `moonbeam-docs` repo to remove images
- [ ] Will need to create PR in `moonbeam-docs` repo to remove variables
- [ ] Will need to create PR in `moonbeam-mkdocs` repo to add redirects for Chinese site
- [x] No additional PRs are required after the translations are done

### CODE REVIEW

File: builders/get-started/eth-compare/security.md [hunk 1]
```diff
@@ -129,4 +129,16 @@ On Ethereum, you can use this check to ensure that a given contract function can
...
+
+[Mintable XC-20s](/builders/interoperability/xcm/xc20/mintable-xc20){target=_blank} are a form of [XC-20s](/builders/interoperability/xcm/xc20/overview){target=_blank} that are minted and burned in Moonbeam directly. Like all XC-20s, Mintable XC-20s are Substrate assets that can be interacted with through an ERC-20 interface via a precompile contract. With Mintable XC-20s specifically, the ERC-20 interface has been extended to include some [additional functionality](/builders/interoperability/xcm/xc20/mintable-xc20/#additional-functions){target=_blank}, such as the ability to mint and burn tokens, freeze and unthaw tokens and accounts, and more. This additional functionality is similar to the standard ERC-20 extensions in Ethereum, like the [ERC20Mintable](https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#ERC20Mintable){target=_blank}, [ERC20Burnable](https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#ERC20Burnable){target=_blank}, and [ERC20Pausable](https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#ERC20Pausable){target=_blank} extensions, but it is important to note that they **are not the exact same**.
```
Comment (albertov19) [inline builders/get-started/eth-compare/security.md:136]:
```suggestion
[Mintable XC-20s](/builders/interoperability/xcm/xc20/mintable-xc20){target=_blank} are a form of [XC-20s](/builders/interoperability/xcm/xc20/overview){target=_blank} that are minted and burned in Moonbeam directly. Like all XC-20s, Mintable XC-20s are Substrate assets that can be interacted with through an ERC-20 interface via a precompile contract. With Mintable XC-20s specifically, the ERC-20 interface has been extended to include some [additional functionality](/builders/interoperability/xcm/xc20/mintable-xc20/#additional-functions){target=_blank}, such as the ability to mint and burn tokens, freeze and unthaw tokens and accounts, and more. This additional functionality is similar to the standard ERC-20 extensions in Ethereum, like the [ERC20Mintable](https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#ERC20Mintable){target=_blank}, [ERC20Burnable](https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#ERC20Burnable){target=_blank}, and [ERC20Pausable](https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#ERC20Pausable){target=_blank} extensions, **but it is important to note that they are not exactly the same**. 
```

File: builders/get-started/eth-compare/security.md [hunk 2]
```diff
@@ -129,4 +129,16 @@ On Ethereum, you can use this check to ensure that a given contract function can
...
+
+The standard Ethereum ERC-20 [`burn` function](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol#L277-L293){target=_blank} requires that the account the tokens are being burned from has at least the requested amount of tokens to burn. Whereas in Substrate, the functionality differs and that requirement does not exist. With that being said, you'll need to manually require that the account has enough tokens, using the [require function](https://docs.soliditylang.org/en/v0.8.17/control-structures.html#panic-via-assert-and-error-via-require){target=_blank}, like so:
```
Comment (albertov19) [inline builders/get-started/eth-compare/security.md:138]:
```suggestion
A common Ethereum ERC-20 [`burn` function](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol#L277-L293){target=_blank} requires that the account the tokens are being burned from has at least the requested amount of tokens to burn.  That is, if a user calls the function trying to burn more tokens that they actually own, the call will fail.

In Substrate, the functionality differs, and that requirement does not exist. Consequently, a user might call the function with a much larger burn amount that what they actually hold, and the call will be successful, but only the amount of tokens they hold will be burned. With that being said, you'll need to manually require that the account has enough tokens, using the [require function](https://docs.soliditylang.org/en/v0.8.17/control-structures.html#panic-via-assert-and-error-via-require){target=_blank}, like so:
```

File: builders/get-started/eth-compare/security.md [hunk 3]
```diff
@@ -129,4 +129,16 @@ On Ethereum, you can use this check to ensure that a given contract function can
...
+
+```
+require(mintableERC20.balanceOf(from) >= value, "burn amount exceeds balance")
+```
+
+Additionally, it is also important to note that for Mintable XC-20s the maximum `value` that can be minted is actually limited to *uint128*. A Mintable XC-20 will behave differently if the total supply is over 2^128 (without decimals), the mint will fail due to overflow checks. This is unlikely to happen for traditional tokens as they are not meant to reach such high numbers, but nonetheless is important to mention as this is different than the standard Ethereum ERC-20.
```
Comment (albertov19) [inline builders/get-started/eth-compare/security.md:144]:
```suggestion
Additionally, it is also important to note that for Mintable XC-20s the maximum `value` (or total supply) that can be minted is actually limited to *uint128*, compared to common mintable ERC-20s where the total supply is capped to *uint256*. A Mintable XC-20 will behave differently if the total supply is over 2^128 (without decimals), the mint will fail due to overflow checks. This is unlikely to happen for traditional tokens as they are not meant to reach such high numbers, but nonetheless is important to mention as this is different than the standard Ethereum ERC-20.
```

File: builders/interoperability/xcm/xc20/mintable-xc20.md [hunk 4]
```diff
@@ -48,7 +48,13 @@ The Solidity interface for Mintable XC-20 tokens is a combination of the followi
 Mintable XC-20s include additional functions that only the owner or the designated account is allowed to call. They are declared in the [LocalAsset.sol](https://github.com/PureStake/moonbeam/blob/master/precompiles/assets-erc20/LocalAsset.sol){target=_blank} interface, and are as follows:
 
 - **mint(*address* to, *uint256* value)** - mints a given amount of tokens to a specified address. Only the owner and the issuer are capable of calling this function
+    !!! note
+    The maximum `value` that can be minted is actually limited to *uint128*. A Mintable XC-20 will behave differently if the total supply is over 2^128 (without decimals), the mint will fail due to overflow checks. This is unlikely to happen for traditional tokens as they are not meant to reach such high numbers.
+
 - **burn(*address* from, *uint256* value)** - burns a given amount of tokens from a specified address. Only the owner and the admin are capable of calling this function
+    !!! note
```
Comment (albertov19) [inline builders/interoperability/xcm/xc20/mintable-xc20.md:55]: Add URL to security check section

File: builders/interoperability/xcm/xc20/mintable-xc20.md [hunk 5]
```diff
@@ -48,7 +48,13 @@ The Solidity interface for Mintable XC-20 tokens is a combination of the followi
 Mintable XC-20s include additional functions that only the owner or the designated account is allowed to call. They are declared in the [LocalAsset.sol](https://github.com/PureStake/moonbeam/blob/master/precompiles/assets-erc20/LocalAsset.sol){target=_blank} interface, and are as follows:
 
 - **mint(*address* to, *uint256* value)** - mints a given amount of tokens to a specified address. Only the owner and the issuer are capable of calling this function
+    !!! note
+    The maximum `value` that can be minted is actually limited to *uint128*. A Mintable XC-20 will behave differently if the total supply is over 2^128 (without decimals), the mint will fail due to overflow checks. This is unlikely to happen for traditional tokens as they are not meant to reach such high numbers.
```
Comment (albertov19) [inline builders/interoperability/xcm/xc20/mintable-xc20.md:52]: Add URL to security check section

**Review by albertov19 — APPROVED** [review]
Approved but left some suggestions

File: builders/interoperability/xcm/xc20/mintable-xc20.md (hunk 5 above)
Comment (eshaben) [inline builders/interoperability/xcm/xc20/mintable-xc20.md:52]:
```suggestion
    The maximum `value` that can be minted is actually limited to *uint128*. A Mintable XC-20 will behave differently if the total supply is over 2^128 (without decimals), the mint will fail due to overflow checks. This is unlikely to happen for traditional tokens as they are not meant to reach such high numbers. For more information, please refer to the [Mintable XC-20s vs ERC-20s](/builders/get-started/eth-compare/security/#mintable-xc-20s-vs-erc-20s){target=_blank} section of the Security Considerations page.
```

File: builders/interoperability/xcm/xc20/mintable-xc20.md [hunk 6]
```diff
@@ -48,7 +48,13 @@ The Solidity interface for Mintable XC-20 tokens is a combination of the followi
...
+
 - **burn(*address* from, *uint256* value)** - burns a given amount of tokens from a specified address. Only the owner and the admin are capable of calling this function
+    !!! note
+    The `burn` function in Substrate behaves differently than the [standard ERC-20 `burn` function](https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#ERC20-_burn-address-uint256-){target=_blank} as it does not require the `from` account to have the amount of tokens specified by the `value`.
```
Comment (eshaben) [inline builders/interoperability/xcm/xc20/mintable-xc20.md:56]:
```suggestion
    The `burn` function in Substrate behaves differently than the [standard ERC-20 `burn` function](https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#ERC20-_burn-address-uint256-){target=_blank} as it does not require the `from` account to have the amount of tokens specified by the `value`. For more information, please refer to the [Mintable XC-20s vs ERC-20s](/builders/get-started/eth-compare/security/#mintable-xc-20s-vs-erc-20s){target=_blank} section of the Security Considerations page.
```

**Pull request merged**